File: replace_headers.py
import os
import re

def get_home_header(index_path):
    with open(index_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Match top-bar and header
    # The header in index.html starts with <div class="top-bar"> and ends with </header>
    pattern = re.compile(
        r'(<div class="top-bar">.*?</header>)',
        re.DOTALL
    )
    match = pattern.search(content)
    if match:
        return match.group(1)
    else:
        raise ValueError("Could not find top-bar and header in index.html")

def adjust_links_for_subdirectories(header_html):
    # This function adjusts links in the header for files that are exactly 1 directory deep
    # (e.g., services/, locations/, resources/)
    
    # Links to index.html
    header_html = re.sub(r'href="index\.html"', 'href="../index.html"', header_html)
    # Links to about.html
    header_html = re.sub(r'href="about\.html"', 'href="../about.html"', header_html)
    # Links to before-after.html
    header_html = re.sub(r'href="before-after\.html"', 'href="../before-after.html"', header_html)
    # Links to reviews.html
    header_html = re.sub(r'href="reviews\.html"', 'href="../reviews.html"', header_html)
    
    # Links to directories
    header_html = re.sub(r'href="services/', 'href="../services/', header_html)
    header_html = re.sub(r'href="locations/', 'href="../locations/', header_html)
    header_html = re.sub(r'href="resources/', 'href="../resources/', header_html)
    
    # In case there are images inside the header
    header_html = re.sub(r'src="Images/', 'src="../Images/', header_html)
    header_html = re.sub(r'src="images/', 'src="../images/', header_html)

    return header_html

def process_files(directory):
    index_path = os.path.join(directory, 'index.html')
    home_header = get_home_header(index_path)
    sub_header = adjust_links_for_subdirectories(home_header)
    
    header_pattern = re.compile(
        r'(<!-- Top Bar -->|<div class="top-bar">).*?(</header>)',
        re.DOTALL
    )

    count = 0
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.html') and file != 'index.html':
                filepath = os.path.join(root, file)
                
                # Check depth
                rel_path = os.path.relpath(filepath, directory)
                depth = rel_path.count(os.sep)
                
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()

                match = header_pattern.search(content)
                if not match:
                    # Some files might not have top-bar, try just header
                    alt_pattern = re.compile(r'<header[^>]*>.*?</header>', re.DOTALL)
                    match = alt_pattern.search(content)
                
                if match:
                    header_to_use = home_header if depth == 0 else sub_header
                    
                    new_content = content[:match.start()] + header_to_use + content[match.end():]
                    
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated header in: {rel_path}")
                    count += 1
                else:
                    print(f"Could not find header in: {rel_path}")
    print(f"Total files updated: {count}")

if __name__ == "__main__":
    process_files('.')

File: replace_headers.js
const fs = require('fs');
const path = require('path');

function getHomeHeader(indexPath) {
    const content = fs.readFileSync(indexPath, 'utf8');
    const pattern = /<div class="top-bar">[\s\S]*?<\/header>/;
    const match = content.match(pattern);
    if (match) {
        return match[0];
    } else {
        throw new Error("Could not find top-bar and header in index.html");
    }
}

function adjustLinksForSubdirectories(headerHtml) {
    headerHtml = headerHtml.replace(/href="index\.html"/g, 'href="../index.html"');
    headerHtml = headerHtml.replace(/href="about\.html"/g, 'href="../about.html"');
    headerHtml = headerHtml.replace(/href="before-after\.html"/g, 'href="../before-after.html"');
    headerHtml = headerHtml.replace(/href="reviews\.html"/g, 'href="../reviews.html"');
    
    headerHtml = headerHtml.replace(/href="services\//g, 'href="../services/');
    headerHtml = headerHtml.replace(/href="locations\//g, 'href="../locations/');
    headerHtml = headerHtml.replace(/href="resources\//g, 'href="../resources/');
    
    headerHtml = headerHtml.replace(/src="Images\//g, 'src="../Images/');
    headerHtml = headerHtml.replace(/src="images\//g, 'src="../images/');
    return headerHtml;
}

function walk(dir, callback) {
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat && stat.isDirectory()) {
            if (file !== '.git' && file !== 'node_modules') {
                walk(filepath, callback);
            }
        } else {
            callback(filepath);
        }
    });
}

function processFiles(directory) {
    const indexPath = path.join(directory, 'index.html');
    const homeHeader = getHomeHeader(indexPath);
    const subHeader = adjustLinksForSubdirectories(homeHeader);
    
    const headerPattern = /(<!-- Top Bar -->|<div class="top-bar">)[\s\S]*?<\/header>/;
    const altPattern = /<header[^>]*>[\s\S]*?<\/header>/;

    let count = 0;
    walk(directory, (filepath) => {
        if (path.extname(filepath) === '.html' && path.basename(filepath) !== 'index.html') {
            const relPath = path.relative(directory, filepath);
            const depth = (relPath.match(/[\\/]/g) || []).length;
            
            let content = fs.readFileSync(filepath, 'utf8');
            let match = content.match(headerPattern);
            if (!match) {
                match = content.match(altPattern);
            }
            
            if (match) {
                const headerToUse = depth === 0 ? homeHeader : subHeader;
                const newContent = content.slice(0, match.index) + headerToUse + content.slice(match.index + match[0].length);
                fs.writeFileSync(filepath, newContent, 'utf8');
                console.log(`Updated header in: ${relPath}`);
                count++;
            } else {
                console.log(`Could not find header in: ${relPath}`);
            }
        }
    });
    console.log(`Total files updated: ${count}`);
}

processFiles('.');
